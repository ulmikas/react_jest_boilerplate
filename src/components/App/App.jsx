import React, { Component } from 'react';
import TabsComponent from '../TabsComponent';
import './App.css';

class App extends Component {
  render() {
    return (<TabsComponent />);
  }
}

export default App;
