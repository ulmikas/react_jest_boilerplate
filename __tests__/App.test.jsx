import React from 'react';
import { mount, shallow, render } from 'enzyme';
import App from '../src/components/App';

describe('<App />', () => {
  it('should render correctly tab 3', () => {
    const tree = mount(<App />);
    const tabLast = tree.find('[data-tabs="tab"]').last();
    tabLast.simulate('click');
    const tabContentLast = tree.find('[data-tabs="tab-panel"]').last();
    expect(tabContentLast).toMatchSnapshot();
  });
});
