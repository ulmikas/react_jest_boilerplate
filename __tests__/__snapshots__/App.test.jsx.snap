// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly tab 3 1`] = `
<div
  data-tabs="tab-panel"
>
  <h2>
    Any content 1
  </h2>
</div>
`;
